engine/userdata/src/main/java/org/apache/cloudstack/userdata/UserDataManagerImpl.java

```diff
@@ -90,49 +88,50 @@ public String concatenateUserData(String userdata1, String userdata2, String use
 
     @Override
     public String validateUserData(String userData, BaseCmd.HTTPMethod httpmethod) {
-        byte[] decodedUserData = null;
-        if (userData != null) {
-
-            if (userData.contains("%")) {
-                try {
-                    userData = URLDecoder.decode(userData, "UTF-8");
-                } catch (UnsupportedEncodingException e) {
-                    throw new InvalidParameterValueException("Url decoding of userdata failed.");
-                }
-            }
+        if (StringUtils.isBlank(userData)) {
```

Comment:
Could you add a log line at the start of the method that logs the userdata that is being validated? something like:
```suggestion
        logger.debug(String.format("Validating userdata [%s].", userData));
        
        if (StringUtils.isBlank(userData)) {
```

Reply:
@JoaoJandre may not be good to log user data here (can be of more length sometimes).

Reply:
@sureshanaparti Maybe leave it at `trace` level? This way it will only show up when someone really needs to see it.

Reply:
we can log it later, may be before validation, in case needed @JoaoJandre I think, logger obj for only this trace is unnecessary.